plugins/videobasedtracker/HDKLedIdentifier.cpp

```diff
@@ -108,26 +105,18 @@ namespace vbtracker {
         // pattern matches any of them.  If so, return that pattern.  We
         // need to check all potential rotations of the pattern, since we
         // don't know when the code started.  For the HDK, the codes are
-        // rotationally invariant.
-
-        /// @todo feels like there should be a good algorithm for
-        /// rotation-invariant string matching besides brute-forcing it here.
-        ///  -- rpavlik
+        // rotationally invariant.  We do this by making wrapped strings
+        // and seeing if the pattern shows up anywhe in them, relying on
+        // the std::string find method to do efficiently.
         for (size_t i = 0; i < d_patterns.size(); i++) {
-            for (size_t j = 0; j < bits.size(); j++) {
-                if (bits == d_patterns[i]) {
-                    return static_cast<int>(i);
-                }
-
-                // So long as we don't find the solution, this rotates
-                // back to the initial configuration after each inner loop.
-                std::list<bool>::iterator mid = bits.begin();
-                std::rotate(bits.begin(), ++mid, bits.end());
-            }
+          if (d_patterns[i].find(bits) != std::string::npos) {
+            return static_cast<int>(i);
+          }
         }
 
         // No pattern recognized and we should have recognized one, so return
-        // a low negative.  We've used -2 so return -3.
+        // a low negative.  We've used -2 for too-small brightness
+        // difference so return -3 for this.
         return -3;
```

Comment:
They are described in the Running.md document.  Added a description of what -2 means, since it was not described.